File: Algorithms/Medium/P2414-Length_of_the_Longest_Alphabetical_Continuous_Substring/p2414.ac.py
class Solution:
    def longestContinuousSubstring(self, s: str) -> int:
        ans: int = 0 if len(s) == 0 else 1;
        tmp_ans: int = ans;
        for i in range(1, len(s)):
            if ord(s[i]) == ord(s[i - 1]) + 1:
                tmp_ans += 1;
                ans = tmp_ans if tmp_ans > ans else ans;
            else:
                tmp_ans = 1;
        return ans;

File: Algorithms/Medium/P2414-Length_of_the_Longest_Alphabetical_Continuous_Substring/p2414.ac.rs
impl Solution {
    pub fn longest_continuous_substring(s: String) -> i32 {
        let mut ans: i32 = if s.len() == 0 { 0 } else { 1 };
        let mut tmp_ans: i32 = ans;
        let arr = s.chars()
                   .map(|char| char as i32 - 96)
                   .collect::<Vec<i32>>();
        for i in 1..arr.len() {
            if arr[i] == arr[i - 1] + 1 {
                tmp_ans = tmp_ans + 1;
                ans = if tmp_ans > ans { tmp_ans } else { ans };
            } else {
                tmp_ans = 1;
            }
        }
        ans
    }
}